volatility3/cli/volshell/generic.py

```diff
@@ -344,67 +362,129 @@ def display_type(
 
             for member in sorted(
                 volobject.vol.members, key=lambda x: (volobject.vol.members[x][0], x)
             ):
                 relative_offset, member_type = volobject.vol.members[member]
                 len_offset = len(hex(relative_offset))
                 len_member = len(member)
-                len_typename = len(member_type.vol.type_name)
+                member_type_name = self._get_type_name_with_pointer(
+                    member_type
+                )  # special case for pointers to show what they point to
+                len_typename = len(member_type_name)
                 if isinstance(volobject, interfaces.objects.ObjectInterface):
                     # We're an instance, so also display the data
                     print(
+                        "    " * dereference_count,
                         " " * (longest_offset - len_offset),
                         hex(relative_offset),
                         ":  ",
                         member,
                         " " * (longest_member - len_member),
                         "  ",
-                        member_type.vol.type_name,
+                        member_type_name,
                         " " * (longest_typename - len_typename),
                         "  ",
                         self._display_value(getattr(volobject, member)),
                     )
                 else:
                     print(
+                        "    " * dereference_count,
                         " " * (longest_offset - len_offset),
                         hex(relative_offset),
                         ":  ",
                         member,
                         " " * (longest_member - len_member),
                         "  ",
-                        member_type.vol.type_name,
+                        member_type_name,
                     )
 
+        else:  # simple type with no members, only one line to print
+            # if the orginal object was just a type string, display the type information
+            if isinstance(object, str):
+                print(self._display_simple_type(volobject, include_value=False))
+
+            # if the original object was an actual volobject or was a type string
+            # with an offset. Then append the actual data to the display.
+            else:
+                print("    " * dereference_count, self._display_simple_type(volobject))
+
+    def _display_simple_type(self, volobject: Any, include_value: bool = True) -> str:
+        # build the display_type_string based on the aviable information
+        if hasattr(volobject.vol, "size"):
+            display_type_string = (
+                f"{volobject.vol.type_name} ({volobject.vol.size} bytes)"
+            )
+        elif hasattr(volobject.vol, "data_format"):
+            data_format = volobject.vol.data_format
+            display_type_string = "{} ({} bytes, {} endian, {})".format(
```

Comment:
Ah, so this came from volshell originally and isn't something I've added here. (I did have to move the functions around though so it looks like its new).

Now I've never actually seen this output in volshell myself, and did think about removing it. I decided to leave it as just because I don't use it - maybe someone out there does. 

I'm happy to put it behind a --verbose flag if that's useful? 

It's currently here if it helps: https://github.com/volatilityfoundation/volatility3/blob/a08b7809b58d3b57a00531b378f7c6f08f33be2e/volatility3/cli/volshell/generic.py#L349-L358

Reply:
Looking into this I see it pop up when looking at just simple types when they're still templates and it seems actually useful. It's not spammed very frequently.

e.g. it's the difference here:

```python
(layer_name) >>> dt('int')  
symbol_table_name1!int (4 bytes, little endian, signed)

(layer_name) >>> dt(task.pid)
 symbol_table_name1!int (4 bytes): 8600
```